HW2/P5/physics.pyx

```diff
@@ -113,17 +145,30 @@ cpdef update(FLOAT[:, ::1] XY,
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads, with static
         # scheduling.
-        for i in range(count):
-            sub_update(XY, V, R, i, count, Grid, grid_spacing)
+        for i in prange(count,num_threads=num_threads,schedule='static', chunksize=count/4):
+        #for i in range(count):
+            sub_update(XY, V, R, i, count, Grid, grid_spacing, locks_ptr)
 
         # update positions
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads (with static
         #    scheduling).
         # SUBPROBLEM 2: update the grid values.
-        for i in range(count):
+        
+        #reset all values in grid
+        for idx_x in prange(grid_size,num_threads=num_threads,schedule='static'):
```

Comment:
This nested loop seems unnecessary since you could achieve the same effect by integrating this into the nested loops below